exchange/order/limits/limits_types.go

```diff
@@ -0,0 +1,74 @@
+package limits
+
+import (
+	"errors"
+	"sync"
+	"time"
+
+	"github.com/thrasher-corp/gocryptotrader/common/key"
+)
+
+// Public errors for order limits
+var (
+	ErrCannotLoadLimit         = errors.New("cannot load limit, levels not supplied")
+	ErrOrderLimitNotFound      = errors.New("order limit not found")
+	ErrExchangeLimitNotLoaded  = errors.New("exchange limits not loaded")
+	ErrPriceBelowMin           = errors.New("price below minimum limit")
+	ErrPriceExceedsMax         = errors.New("price exceeds maximum limit")
+	ErrPriceExceedsStep        = errors.New("price exceeds step limit") // price is not divisible by its step
+	ErrAmountBelowMin          = errors.New("amount below minimum limit")
+	ErrAmountExceedsMax        = errors.New("amount exceeds maximum limit")
+	ErrAmountExceedsStep       = errors.New("amount exceeds step limit") // amount is not divisible by its step
+	ErrNotionalValue           = errors.New("total notional value is under minimum limit")
+	ErrMarketAmountBelowMin    = errors.New("market order amount below minimum limit")
+	ErrMarketAmountExceedsMax  = errors.New("market order amount exceeds maximum limit")
+	ErrMarketAmountExceedsStep = errors.New("market order amount exceeds step limit") // amount is not divisible by its step for a market order
+)
+
+var (
+	errExchangeNameEmpty   = errors.New("exchange name is empty")
+	errAssetInvalid        = errors.New("asset is invalid")
+	errPairNotSet          = errors.New("currency pair is not set")
+	errInvalidPriceLevels  = errors.New("invalid price levels, cannot load limits")
+	errInvalidAmountLevels = errors.New("invalid amount levels, cannot load limits")
+	errInvalidQuoteLevels  = errors.New("invalid quote levels, cannot load limits")
+)
+
+// executionLimits defines minimum and maximum values in relation to
+// order size, order pricing, total notional values, total maximum orders etc
+// for execution on an exchange.
```

Comment:
🦍 Think we can make this briefer
```suggestion
// executionLimits defines minimum and maximum values for order size, pricing, max orders, etc
```